src/RoslynPad/DocumentViewModel.cs

```diff
@@ -19,16 +19,76 @@ internal class DocumentViewModel : NotificationObject
         private DocumentViewModel(MainViewModel mainViewModel)
         {
             MainViewModel = mainViewModel;
-            var defaultPath = GetDefaultPath();
-            Directory.CreateDirectory(defaultPath);
-            Path = defaultPath;
+            Path = GetUserDocumentPath();
+            Directory.CreateDirectory(Path);
             IsFolder = true;
         }
 
-        public string Path { get; }
+        public string Path { get; internal set; }
```

Comment:
Remove this setter. We'll simply create a new `DocumentRoot` when changing folders.